### DISCUSSION TITLE
yoloe data validation

### DISCUSSION THREAD

**Discussion by ahamedistiaque** [ultralytics/ultralytics#22272] [Q&A] [answered]
Hi Team,

I am training YOLOE models for object detection but starting from segmentation pretrained weights " yoloe-11m-seg.pt" also tested with yoloe-v8s/m/l-seg.pt and yoloe-11s/m/l-seg.pt) which training on my detection-only dataset works fine, but validation fails when I try to run directly with the (-seg) model:

from ultralytics import YOLOE
data_path = "/content/data.yaml"
model = YOLOE("yoloe-11m-seg.pt")
metrics = model.val(
        data=data_path,
        imgsz=640,         
        device=0,          
        task="detect",     
        name="val_detect",
    )
print("Box mAP50-95:", metrics.box.map)
print("Segmentation mAP50-95:", metrics.seg.map)  # fails (no masks)

I understand that -seg weights expect a segmentation dataset, but my dataset has only bounding box annotations.

My questions:

1. What’s the recommended way to validate object detection metrics when using YOLOE  pretrained weights on a detection-only dataset?

2. Can I safely evaluate detection metrics (mAP@0.5, Precision, Recall) with a (~seg) model if the dataset has no masks?

3. If I want to use -seg pretrained weights but only have detection annotations, how should I configure val? Any examples for ignoring masks during evaluation?

Thanks a lot for clarifying! 
<img width="451" height="113" alt="Screenshot 2025-10-01 145231" src="https://github.com/user-attachments/assets/b635d2a3-a436-4a3b-9ad1-3b2415d8fd9a" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ahamedistiaque, thanks for opening this discussion and for your interest in Ultralytics 🚀! This is an automated response to help you get started, and an Ultralytics engineer will follow up with you shortly.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including:
-Your exact code snippet used for training/validation
-The `data.yaml` you’re using
-Environment details (OS, Python, PyTorch, CUDA), plus `pip show ultralytics`
-Model weights and task configuration you’re running
-Console logs and any screenshots of errors
-And verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

To help us assist with your validation questions, please clarify whether this is a bug or a usage question and include the details listed above. Once we have that, we’ll take a closer look and guide you accordingly 🔍💡

**Reply by glenn-jocher** [reply, marked as answer]
Thanks for the clear questions. Passing `task='detect'` to a `*-seg.pt` keeps the segmentation validator, so `metrics.seg.*` expects masks; for detection-only data use one of these:

```python
# A) Recommended: init a detection model, then load the seg checkpoint
from ultralytics import YOLOE

det = YOLOE("yoloe-11m.yaml")
det.load("yoloe-11m-seg.pt")
metrics = det.val(data=data_path, imgsz=640)
print(metrics.box.map, metrics.box.map50, metrics.box.map75)
```

```python
# B) Keep the seg checkpoint but force the detection validator
from ultralytics import YOLOE
from ultralytics.models.yolo.yoloe import YOLOEDetectValidator

seg = YOLOE("yoloe-11m-seg.pt")
metrics = seg.val(data=data_path, imgsz=640, validator=YOLOEDetectValidator)
print(metrics.box.map)
```

This is safe: you can evaluate mAP/Precision/Recall for detection with a `*-seg.pt` model on a bbox-only dataset, but don’t access `metrics.seg.*` since no masks exist. See the detection-from-seg weights example in the YOLOE docs and validator details in the references: [YOLOE training and inference](https://docs.ultralytics.com/models/yoloe/), [YOLOE validators reference](https://docs.ultralytics.com/reference/models/yolo/yoloe/val/), and the [Model.val reference](https://docs.ultralytics.com/reference/engine/model/#val). If you still see a validator mismatch, please share the full traceback and the text output of `yolo checks`.
